### PR TITLE
Feat(compiler-base-session): Add some methods for Session.

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

issue #115 

#### 2. What is the scope of this PR (e.g. component or file name):

compiler-base-session

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [x] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

The old Session provide 'emit_err' which will panic when the diagnostic information is emitted. This situation is very rare in actual usage scenarios, so this PR provides some interfaces for adding errors/warnings, emitting diagnostics, and aborting the program.

'emit_stashed_diagnostics_and_abort' to emit all the diagnostics and abort.
'emit_stashed_diagnostics' to emit all the diagnostics but not abort.
'add_err' to add error to diagnostics.
'add-warn' to add warning to diagnostics.
'diagnostic_count' to get the count of diagnostics.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Peefy — APPROVED** [review]
LGTM

**Comment by Peefy** [comment]
The reason for code modification should be clearly described in PR or code comments.

File: compiler_base/session/src/lib.rs [hunk 1]
```diff
@@ -162,19 +162,176 @@ impl Session {
...
+    ///    // 4. Add the error diagnostic.
+    ///    sess.add_err(MyError {}).unwrap();
+    ///    // 5. Emit the error diagnostic.
+    ///    sess.emit_stashed_diagnostics_and_abort().unwrap();
     /// });
     /// assert!(result.is_err());
-    ///
     /// ```
-    pub fn emit_err(&self, err: impl SessionDiagnostic) -> Result<bool> {
+    pub fn emit_stashed_diagnostics_and_abort(&self) -> Result<bool> {
```
Comment (Peefy) [inline compiler_base/session/src/lib.rs:172]: Maybe use `!` type would be better.

Comment (zong-zhe) [reply compiler_base/session/src/lib.rs]: fixed.

**Review by Peefy — APPROVED** [review]
LGTM

**Review by amyXia1994 — APPROVED** [review]

**Pull request merged**